### PR TITLE
Refactor batch of handlers

### PR DESCRIPTION
**Pull request by dpgraham**
Going to make the transition to Kotlin a little bit cleaner by only changing a handful of files at a time.

### CODE REVIEW

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt [hunk 1]
```diff
@@ -14,31 +14,32 @@
...
-        String waitActivityName = params.getDesiredCapabilities().getAppWaitActivity();
-        Long appWaitDuration = params.getDesiredCapabilities().getAppWaitDuration();
+    @Throws(AppiumException::class)
+    override fun handle(params: SessionParams): Session {
+        val appiumSession = Session.createGlobalSession(params.desiredCapabilities)
+        val activityName = params.desiredCapabilities.appActivity
+        val waitActivityName = params.desiredCapabilities.appWaitActivity
+        val appWaitDuration = params.desiredCapabilities.appWaitDuration
         try {
             if (activityName != null) {
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt]: use `let`

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/DrawerActionHandler.kt [hunk 2]
```diff
@@ -14,46 +14,42 @@
...
-    @Override
-    public Void handle(DrawerActionParams params) throws AppiumException {
-        ViewInteraction viewInteraction = Element.getViewInteractionById(params.getElementId());
-        Integer gravity = params.getGravity();
+    @Throws(AppiumException::class)
+    override fun handle(params: DrawerActionParams): Void? {
+        val viewInteraction = Element.getViewInteractionById(params.elementId)
+        val gravity = params.gravity
         try {
             if (isOpenAction) {
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/DrawerActionHandler.kt]: I'd rather change this to:

```
val action = gravity?.let { () -> DrawerActions.open(it) if isOpenAction else () -> DrawerActions.close(ot) } 
  ?: (() -> DrawerActions.open() if isOpenAction else () -> DrawerActions.close())
viewInteraction.perform(action)
```

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/ElementEquals.java [hunk 3]
```diff
@@ -1,22 +0,0 @@
-package io.appium.espressoserver.lib.handlers;
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/ElementEquals.java]: is it not used?

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/ElementValue.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                view.value = Integer.parseInt(params.value)
+                return null
+            }
+        } catch (e: NumberFormatException) {
+            throw InvalidArgumentException(String.format("Cannot convert '%s' to an integer",
+                    params.value))
+        }
+
+        val viewInteraction = Element.getViewInteractionById(elementId)
+        if (isReplacing) {
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/ElementValue.kt]: viewInteraction.perform(replaceText(params.value) if isReplacing else typeText(params.value))

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/FindElement.kt [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import io.appium.espressoserver.lib.model.Locator
+import io.appium.espressoserver.lib.viewaction.ViewGetter
+
+import io.appium.espressoserver.lib.helpers.ViewFinder.findBy
+
+class FindElement : RequestHandler<Locator, Element> {
+
+    @Throws(AppiumException::class)
+    override fun handle(locator: Locator): Element {
+        var parentView: View? = null
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/FindElement.kt]: use ?.let

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/FindElements.kt [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+import io.appium.espressoserver.lib.helpers.ViewFinder.findAllBy
+
+class FindElements : RequestHandler<Locator, List<Element>> {
+
+    @Throws(AppiumException::class)
+    override fun handle(locator: Locator): List<Element> {
+        var parentView: View? = null
+        if (locator.elementId != null) {
+            parentView = ViewGetter().getView(Element.getViewInteractionById(locator.elementId))
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/FindElements.kt]: use ?.let

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/FindElements.kt [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        }
+        if (locator.using == null) {
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/FindElements.kt]: Would elvis operator (?:) be more readable here?

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/FindElements.kt [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+            throw InvalidStrategyException("Locator strategy cannot be empty")
+        } else if (locator.value == null) {
+            throw MissingCommandsException("No locator provided")
+        }
+
+        // Get the viewInteractions
+        val views = findAllBy(parentView,
+                locator.using, locator.value)
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/FindElements.kt]: there's np need to wrap the args

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/FindElements.kt [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+        // Turn it into a list of elements
+        val elements = ArrayList<Element>()
+        for (view in views) {
+            elements.add(Element(view))
+        }
+
+        // If we have a match, return success
+        return elements
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/FindElements.kt]: use map

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/ElementValue.kt (hunk 4 above)
Comment (dpgraham) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/ElementValue.kt]: Are you sure that's valid Kotlin syntax. I tried copy-and-pasting and it didn't work.

**Comment by dpgraham** [comment]
Fixes pushed

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/DrawerActionHandler.kt [hunk 10]
```diff
@@ -28,20 +28,11 @@ class DrawerActionHandler(private val isOpenAction: Boolean) : RequestHandler<Dr
...
-                } else {
-                    viewInteraction.perform(DrawerActions.open(gravity))
-                }
-            } else {
-                if (gravity == null) {
-                    viewInteraction.perform(DrawerActions.close())
-                } else {
-                    viewInteraction.perform(DrawerActions.close(gravity))
-                }
+            params.gravity?.let {
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/DrawerActionHandler.kt]: 👍

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/ElementValue.kt (hunk 4 above)
Comment (mykola-mokhnach) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/ElementValue.kt]: nwm, this is fine

**Pull request merged**